### PR TITLE
Samples for tracing

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/13882

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/samples/sample_servicebus.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# Simple console exporter
+exporter = ConsoleSpanExporter()
+
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+trace.get_tracer_provider().add_span_processor(
+    SimpleExportSpanProcessor(exporter)
+)
+
+# Example with Storage SDKs
```
Comment (rakshith91) [inline sdk/core/azure-core-tracing-opentelemetry/samples/sample_servicebus.py:41]:
```suggestion
# Example with Servicebus SDKs
```

File: sdk/core/azure-core-tracing-opentelemetry/samples/sample_eventgrid.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from azure.core.settings import settings
+from azure.core.tracing.ext.opentelemetry_span import OpenTelemetrySpan
+
+settings.tracing_implementation = OpenTelemetrySpan
+
+# In the below example, we use a simple console exporter, uncomment these lines to use
+# the Azure Monitor Exporter.
+# Example of Azure Monitor exporter, but you can use anything OpenTelemetry supports
+# from azure_monitor import AzureMonitorSpanExporter
+# exporter = AzureMonitorSpanExporter(
```
Comment (lmazuel) [inline sdk/core/azure-core-tracing-opentelemetry/samples/sample_eventgrid.py:22]: Please add a ref to the package name on PyPI as well, so people know what they need to install.

File: sdk/core/azure-core-tracing-opentelemetry/samples/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+products:
+  - azure
+  - azure-core-tracing-opentelemetry
+urlFragment: azure-core-tracing-opentelemetry-samples
+---
+
+# Azure Core Tracing Opentelemetry Integration Python Samples
+
+These code samples show using the tracing opentelemetry integration with few services including servicebus, storage and eventgrid. A simple console span exporter is used in this example to export the calls. However, the code to use the azure monitor exporter has also been provided and commented in each sample.
+
```
Comment (lmazuel) [inline sdk/core/azure-core-tracing-opentelemetry/samples/README.md:14]: We should have a note on the packages needed to be installed in order to have those samples working.

File: sdk/core/azure-core-tracing-opentelemetry/samples/sample_storage.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+trace.get_tracer_provider().add_span_processor(
+    SimpleExportSpanProcessor(exporter)
+)
+
+# Example with Storage SDKs
+
+from azure.storage.blob import BlobServiceClient
+
+with tracer.start_as_current_span(name="MyApplication"):
+    client = BlobServiceClient.from_connection_string('connectionstring')
```
Comment (lmazuel) [inline sdk/core/azure-core-tracing-opentelemetry/samples/sample_storage.py:47]: Please externalize the connectionstring in a variable like you did in previous samples

File: sdk/core/azure-core-tracing-opentelemetry/samples/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+* Trace Storage Calls to create a container: [sample_storage.py][python-sample-storage]
+* Trace Eventgrid calls to publish an event : [sample_eventgrid.py][python-sample-eventgrid]
+* Trace Servicebus calls to send messages to queue : [sample_servicebus.py][python-sample-servicebus]
+
+It is assumed that the relevant SDKs are installed along with this extension. Below are the relevant packages which can be found in Pypi.
```
Comment (lmazuel) [inline sdk/core/azure-core-tracing-opentelemetry/samples/README.md:19]:
```suggestion
It is assumed that the relevant SDKs are installed along with this extension. Below are the relevant packages which can be found in PyPI.
```

**Review by lmazuel — APPROVED** [review]

**Pull request merged**